File: 0206-reverse-linked-list/0206-reverse-linked-list.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        var nodes = head
        
        var values = [Int?]()
        
        while true {
            values.append(nodes?.val)
            
            if nodes?.next == nil {
                break
            }
            nodes = nodes?.next
        }
        
        var rails = head
        while rails != nil {
            rails?.val = values.removeLast()!
            rails = rails?.next
        }
        
        return head
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function reverseList(head: ListNode | null): ListNode | null {
    let current = null;

    while (head !== null) {
        const newNode = new ListNode(head.val);
        newNode.next = current;
        current = newNode;
        head = head.next;
    }

    return current;
};